torch_xla_py/xla_model.py

```diff
@@ -141,28 +141,41 @@ def __init__(self, smooth_factor=0.8):
     self._smooth_factor = smooth_factor
     self._start_time = time.time()
     self._partial_time = self._start_time
-    self._count = 0
-    self._rate = 0.0
+    self._partial_count = 0
+    self._partial_rate = None
+    self._count = 0.0
 
-  def update(self, count):
-    now = time.time()
-    delta = now - self._partial_time
-    if delta > 0:
-      rate = (count - self._count) / delta
-      self._rate = (
-          self._rate * self._smooth_factor + rate * (1.0 - self._smooth_factor))
+  def _update(self, now, rate):
+    self._partial_count += self._count
+    self._count = 0
```

Comment:
On line 146, you use a float for _count. Here, you use an int. Probably best to use an int here as well since you are using count/delta on line 178 and this won't work well on python2 if both count and delta are ints

Reply:
made everything float per suggestion (synced offline)